### ISSUE TITLE
Add executor to the Promise constructor

### ISSUE THREAD

**Issue by vincefliek** [slavik57/promise-sync#13]
Hi! Thanks for your great and helpful lib.

I was trying to do the following, but there wasn't a sync behaviour in my tests:

``` javascript
import { PromiseMock } from 'promise-sync';

const promise = new PromiseMock((res, rej) => {
  someAsyncFunction(
     result => res(result),
     error => rej(error),
   );
});
```

Could you add a support of [constructor executor](https://tc39.github.io/ecma262/#sec-promise-executor), please?

BTW, I will be happy to make a PR and add constructor executor, if you don't mind.

**Comment by slavik57** [comment]
If the `someAsyncFunction` is really async (returns a real promise) there is nothing that can be done to make it synchronous.

Why do you need the constructor executor in the mock?
It can be implemented as follows:
```
var promise = new PromiseMock();
someAsyncFunction(
    result => promise.resolve(result),
    error => promise.reject(error)
);
```

**Comment by vincefliek** [comment]
Sorry, but I am not able to use your approach.

This example of class maybe will be more full

``` javascript
import MyPromise from '../promise';
import someAsyncFunction from '../asyncFn';

class SomeClass {
  ...
  getCurrentLocation = () => new MyPromise((resolve, reject) => {
    someAsyncFunction(
      pos => resolve(pos),
      error => reject(error),
    );
  });
  ...
}
```

Then in my integration tests I mock MyPromise and someAsyncFunction with their sync versions. Right now, the callback in this.getCurrentLocation().then(() => { /* this is success callback */ }) doesn't execute and my tests fail.
Also, I've noticed that if I mock the global Promise object (e.g. window.Promise = PromiseMock), my tests just hang and never finish until I kill the process (with Ctrl+C in cli).

**Comment by slavik57** [comment]
Published under version `1.1.7`

**Comment by vincefliek** [comment]
@slavik57 thanks for extremely quick fix!
